### PR TITLE
gupnp-tools: declare indirect deps with linkage

### PR DESCRIPTION
**Pull request by chenrui333**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
https://github.com/Homebrew/homebrew-core/actions/runs/10343031007/job/28626967843#step:3:209

### CODE REVIEW

**Comment by carlocab** [comment]
Duplicate of #180934?

**Comment by p-linnane** [comment]
Based off the branch name this was probably supposed to be a PR for `gupnp`.

**Comment by chenrui333** [comment]
> Duplicate of #180934?

I am sorry, I dont know how that happened (probably on two screens)

**Comment by chenrui333** [comment]
> https://github.com/Homebrew/homebrew-core/actions/runs/10343031007/job/28626967843#step:3:209

@p-linnane said right, that also points to gupnp-tools

File: Formula/g/gupnp-tools.rb [hunk 1]
```diff
@@ -31,7 +31,11 @@ class GupnpTools < Formula
   depends_on "libxml2"
 
   on_macos do
+    depends_on "at-spi2-core"
+    depends_on "cairo"
     depends_on "gettext"
+    depends_on "harfbuzz"
+    depends_on "pango"
   end
```
Comment (carlocab) [inline Formula/g/gupnp-tools.rb:39]: Don't see linkage to most of these
```
==> brew linkage --cached gupnp-tools
System libraries:
  /usr/lib/libSystem.B.dylib
Homebrew libraries:
  /opt/homebrew/opt/gdk-pixbuf/lib/libgdk_pixbuf-2.0.0.dylib (gdk-pixbuf)
  /opt/homebrew/opt/gettext/lib/libintl.8.dylib (gettext)
  /opt/homebrew/opt/glib/lib/libgio-2.0.0.dylib (glib)
  /opt/homebrew/opt/glib/lib/libglib-2.0.0.dylib (glib)
  /opt/homebrew/opt/glib/lib/libgobject-2.0.0.dylib (glib)
  /opt/homebrew/opt/gssdp/lib/libgssdp-1.6.0.dylib (gssdp)
  /opt/homebrew/opt/gtk+3/lib/libgdk-3.0.dylib (gtk+3)
  /opt/homebrew/opt/gtk+3/lib/libgtk-3.0.dylib (gtk+3)
  /opt/homebrew/opt/gtksourceview4/lib/libgtksourceview-4.0.dylib (gtksourceview4)
  /opt/homebrew/opt/gupnp/lib/libgupnp-1.6.0.dylib (gupnp)
  /opt/homebrew/opt/gupnp-av/lib/libgupnp-av-1.0.3.dylib (gupnp-av)
  /opt/homebrew/opt/libsoup/lib/libsoup-3.0.0.dylib (libsoup)
  /opt/homebrew/opt/libxml2/lib/libxml2.2.dylib (libxml2)
Dependencies with no linkage:
  at-spi2-core
```

Comment (p-linnane) [reply Formula/g/gupnp-tools.rb:39]: I really think it's time for a DSL improvement where we can specify if a dependency is necessary, but won't have linkage.

Comment (chenrui333) [reply Formula/g/gupnp-tools.rb:39]: looks like linkage reporting is not consistent either?

image

Comment (chenrui333) [reply Formula/g/gupnp-tools.rb:39]: this is the linkage check in my local

```
$ brew linkage --cached gupnp-tools
System libraries:
  /System/Library/Frameworks/Carbon.framework/Versions/A/Carbon
  /System/Library/Frameworks/Cocoa.framework/Versions/A/Cocoa
  /System/Library/Frameworks/CoreGraphics.framework/Versions/A/CoreGraphics
  /usr/lib/libSystem.B.dylib
Homebrew libraries:
  /opt/homebrew/opt/at-spi2-core/lib/libatk-1.0.0.dylib (at-spi2-core)
  /opt/homebrew/opt/cairo/lib/libcairo-gobject.2.dylib (cairo)
  /opt/homebrew/opt/cairo/lib/libcairo.2.dylib (cairo)
  /opt/homebrew/opt/gdk-pixbuf/lib/libgdk_pixbuf-2.0.0.dylib (gdk-pixbuf)
  /opt/homebrew/opt/gettext/lib/libintl.8.dylib (gettext)
  /opt/homebrew/opt/glib/lib/libgio-2.0.0.dylib (glib)
  /opt/homebrew/opt/glib/lib/libglib-2.0.0.dylib (glib)
  /opt/homebrew/opt/glib/lib/libgmodule-2.0.0.dylib (glib)
  /opt/homebrew/opt/glib/lib/libgobject-2.0.0.dylib (glib)
  /opt/homebrew/opt/gssdp/lib/libgssdp-1.6.0.dylib (gssdp)
  /opt/homebrew/opt/gtk+3/lib/libgdk-3.0.dylib (gtk+3)
  /opt/homebrew/opt/gtk+3/lib/libgtk-3.0.dylib (gtk+3)
  /opt/homebrew/opt/gtksourceview4/lib/libgtksourceview-4.0.dylib (gtksourceview4)
  /opt/homebrew/opt/gupnp/lib/libgupnp-1.6.0.dylib (gupnp)
  /opt/homebrew/opt/gupnp-av/lib/libgupnp-av-1.0.3.dylib (gupnp-av)
  /opt/homebrew/opt/harfbuzz/lib/libharfbuzz.0.dylib (harfbuzz)
  /opt/homebrew/opt/libsoup/lib/libsoup-3.0.0.dylib (libsoup)
  /opt/homebrew/opt/libxml2/lib/libxml2.2.dylib (libxml2)
  /opt/homebrew/opt/pango/lib/libpango-1.0.0.dylib (pango)
  /opt/homebrew/opt/pango/lib/libpangocairo-1.0.0.dylib (pango)
```

Comment (cho-m) [reply Formula/g/gupnp-tools.rb:39]: Meson defaults to `-dead_strip_dylibs` so if they aren't showing up anymore in CI, then linkage may have been pruned out.

Note that Sonoma was originally using the same buggy `ld` that Ventura is currently using. So old Sonoma bottles will be overlinked. If you want to check linkage, you should either check Monterey bottles, or Sonoma bottles built with Xcode 15.3

**Comment by cho-m** [comment]
Just went ahead and rebuilt the bottles. Superseded by #183521

```
❯ brew linkage gupnp-tools
System libraries:
  /usr/lib/libSystem.B.dylib
Homebrew libraries:
  /opt/homebrew/opt/gdk-pixbuf/lib/libgdk_pixbuf-2.0.0.dylib (gdk-pixbuf)
  /opt/homebrew/opt/gettext/lib/libintl.8.dylib (gettext)
  /opt/homebrew/opt/glib/lib/libgio-2.0.0.dylib (glib)
  /opt/homebrew/opt/glib/lib/libglib-2.0.0.dylib (glib)
  /opt/homebrew/opt/glib/lib/libgobject-2.0.0.dylib (glib)
  /opt/homebrew/opt/gssdp/lib/libgssdp-1.6.0.dylib (gssdp)
  /opt/homebrew/opt/gtk+3/lib/libgdk-3.0.dylib (gtk+3)
  /opt/homebrew/opt/gtk+3/lib/libgtk-3.0.dylib (gtk+3)
  /opt/homebrew/opt/gtksourceview4/lib/libgtksourceview-4.0.dylib (gtksourceview4)
  /opt/homebrew/opt/gupnp/lib/libgupnp-1.6.0.dylib (gupnp)
  /opt/homebrew/opt/gupnp-av/lib/libgupnp-av-1.0.3.dylib (gupnp-av)
  /opt/homebrew/opt/libsoup/lib/libsoup-3.0.0.dylib (libsoup)
  /opt/homebrew/opt/libxml2/lib/libxml2.2.dylib (libxml2)
```

**Pull request closed without merging**